src/auth/auth-config.ts

```diff
@@ -599,6 +631,41 @@ export class MultiFactorAuthConfig implements MultiFactorConfig {
         }
       });
     }
+
+    if (typeof options.providerConfigs !== 'undefined') {
+      if (!validator.isArray(options.providerConfigs)) {
+        throw new FirebaseAuthError(
+          AuthClientErrorCode.INVALID_CONFIG,
+          '"MultiFactorConfig.providerConfigs" must be an array of valid "MultiFactorProviderConfigs."',
+        );
+      }
+      //Validate content of array.
+      options.providerConfigs.forEach((multiFactorProviderConfig) => {
+        if (typeof multiFactorProviderConfig === 'undefined') {
+          throw new FirebaseAuthError(
+            AuthClientErrorCode.INVALID_CONFIG,
+            `"${multiFactorProviderConfig}" is not a valid "MultiFactorProviderConfigType".`
+          )
+        }
+        if (typeof multiFactorProviderConfig.state !== 'undefined' &&
+          multiFactorProviderConfig.state !== 'ENABLED' &&
+          multiFactorProviderConfig.state !== 'DISABLED') {
+          throw new FirebaseAuthError(
+            AuthClientErrorCode.INVALID_CONFIG,
+            '"MultiFactorConfig.providerConfigs.state" must be either "ENABLED" or "DISABLED".',
+          )
+        }
+        if (typeof multiFactorProviderConfig.totpProviderConfig !== 'undefined') {
```

Comment:
not specifying totpProviderConfig is an error right now, since we don't have any other options in MultiFactorProviderConfig.